tools/codegen/gen_lazy_tensor.py

```diff
@@ -68,12 +69,12 @@ def main() -> None:
 
     run(options.source_yaml, options.output_dir, options.dry_run, options.impl_path,
         options.gen_ts_lowerings, options.node_base, options.node_base_hdr,
-        options.tensor_class, options.tensor_class_hdr)
+        options.tensor_class, options.tensor_class_hdr, TSLazyIR)
```

Comment:
One nit is that XLA will also use the default LazyIR (now named TSLazyIR) although they don't actually use TS.  They simply don't generate lowerings, since theirs are handwritten.  The intent was they would omit the --gen_ts_lowerings flag.

In your case, how do you intend to hook up the {Custom}LazyIR class you provided support for?

Reply:
> One nit is that XLA will also use the default LazyIR (now named TSLazyIR) although they don't actually use TS

I'm thinking that they'll need to implement their own subclass (perhaps called `XlaLazyIR`) where the lowering body method does nothing.

> They simply don't generate lowerings, since theirs are handwritten. The intent was they would omit the --gen_ts_lowerings flag.

Ah, I was wondering what that flag was for.

> how do you intend to hook up the {Custom}LazyIR class you provided support for?

My intent was to create a `generate_code.py` python script to invoke the code generation and call `gen_lazy_tensor.run` directly instead of via the CLI. This way, I can pass in a custom `LazyIR` subclass that implements my custom lowering code generation.

> this would also need to be landed to pytorch master.

ah, I see. In which case, would you recommend that I base this branch off of PyTorch `master` instead (and then it will be pulled in the the `lazy_tensor_staging` branch is synced up with latest `master`)? or keep it as is and create a separate PR porting the changes from `lazy_tensor_staging` to `master`?

Reply:
it can land here first and we can merge it there later, either way is ok, most important is to iron out any issues.  I wanted to let @bdhirsh review.

Reply:
@bdhirsh a gentle reminder here to please review when you get a chance